### PR TITLE
Spring Boot Data JPA

### PR DESCRIPTION
**Pull request by VasylAzarov**
added and set Liquibase support | implemented JPARepository support, removed BookRepositoryImpl | added softDelete for Book entity | add book update function (endpoint, service method)

### CODE REVIEW

File: src/main/java/dev/vasyl/proj/controller/BookController.java [hunk 1]
```diff
@@ -33,4 +34,10 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
+
+    @PutMapping("/{id}")
+    public BookDto updateBookById(@PathVariable Long id,
+                                  @RequestBody CreateBookRequestDto requestDto) {
+        return bookService.update(id, requestDto);
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/controller/BookController.java:65]: Check task description and add `DELETE /api/books/{id} (Delete a specific book)`

File: src/main/java/dev/vasyl/proj/model/Book.java [hunk 2]
```diff
@@ -7,21 +7,15 @@
...
-import lombok.Builder;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.Setter;
-import lombok.ToString;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/model/Book.java]:
```suggestion
@Getter
@Setter
```

File: src/main/java/dev/vasyl/proj/model/Book.java [hunk 3]
```diff
@@ -41,5 +35,9 @@ public class Book {
 
     private String description;
 
+    @Column(name = "cover_image")
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/model/Book.java]:
```suggestion
```

File: src/main/java/dev/vasyl/proj/service/BookServiceImpl.java [hunk 4]
```diff
@@ -34,4 +34,14 @@ public BookDto findById(long id) {
                 () -> new EntityNotFoundException("Can`t find Book entity by id: " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find Book entity by id: " + id));
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/service/BookServiceImpl.java]: Create method in mapper
```suggestion
        bookMapper.updateBookFromDto(requestDto, book);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Implement delete method and fix comments

File: src/main/java/dev/vasyl/proj/controller/BookController.java (hunk 1 above)
Comment (VasylAzarov) [reply src/main/java/dev/vasyl/proj/controller/BookController.java:42]: In task description i don't have this api. It's task mistake?


"As a result of this HW you should have the following endpoints in the BookController:

Book Endpoints:

GET: /api/books (Retrieve book catalog) (should be done in the previous PR)
GET: /api/books/{id} (Retrieve book details) (should be done in the previous PR)
POST: /api/books (Create a new book) (should be done in the previous PR)
PUT: /api/books/{id} (Update a specific book)"

File: src/main/java/dev/vasyl/proj/mapper/BookMapper.java [hunk 5]
```diff
@@ -10,9 +10,15 @@
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
 
-    BookDto toDto(Book book);
-
     Book toModel(CreateBookRequestDto createBookRequestDto);
 
+    default Book toModel(CreateBookRequestDto createBookRequestDto, Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/mapper/BookMapper.java]: remove 
don't set id

File: src/main/java/dev/vasyl/proj/model/Book.java [hunk 6]
```diff
@@ -7,21 +7,17 @@
...
-@Table(name = "books")
 @Getter
 @Setter
-@Builder
-@NoArgsConstructor
-@AllArgsConstructor
-@ToString
+@Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/model/Book.java:19]: check if `@Where` is deprecated or not

File: src/main/java/dev/vasyl/proj/service/BookServiceImpl.java [hunk 7]
```diff
@@ -34,4 +34,18 @@ public BookDto findById(long id) {
                 () -> new EntityNotFoundException("Can`t find Book entity by id: " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find Book entity by id: " + id));
+        Book book = bookMapper.toModel(requestDto, id);
```
Comment (Elena-Bruyako) [inline src/main/java/dev/vasyl/proj/service/BookServiceImpl.java]: create method in mapper
`void updateBookFromDto(CreateBookRequestDto requestDto, @MappingTarget Book entity);`
```suggestion
        bookMapper.updateBookFromDto(requestDto, book);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/dev/vasyl/proj/model/Book.java (hunk 6 above)
Comment (VasylAzarov) [reply src/main/java/dev/vasyl/proj/model/Book.java:19]: image

not deprecated

File: src/main/java/dev/vasyl/proj/mapper/BookMapper.java [hunk 8]
```diff
@@ -6,13 +6,23 @@
...
 
-    List<BookDto> toBookDtoList(List<Book> books);
+    void updateBookFromDto(CreateBookRequestDto requestDto, @MappingTarget Book entity);
+
+    BookDto toDto(Book book);
+
+    default Page<BookDto> toBookDtoPage(Page<Book> bookPage) {
+        List<BookDto> dtoList = bookPage.getContent().stream()
+                .map(this::toDto)
+                .toList();
```
Comment (liliia-ponomarenko) [inline src/main/java/dev/vasyl/proj/mapper/BookMapper.java:25]: you can remove this method and use the method map of Page

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml:49]: add one empty line at the end of the file (check other files)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! But fix my comments before merging ;)

**Review by liliia-ponomarenko — APPROVED** [review]

**Pull request merged**